tests/common/auth_test.go

```diff
@@ -0,0 +1,1303 @@
+				password: "user1-123",
+				role:     "role1",
+				key:      "k1",
+				end:      "k3",
+			},
+			{
+				name:     "user2",
+				password: "user2-123",
+				role:     "role2",
+				key:      "k2",
+				end:      "k4",
+			},
+		}
+		for _, user := range users {
+			authSetupTestUser(ctx, cc, user.name, user.password, user.role, user.key, user.end, t)
+		}
+		authEnable(ctx, cc, t)
+		user1c := testutils.MustClient(clus.Client(WithAuth("user1", "user1-123")))
+		user2c := testutils.MustClient(clus.Client(WithAuth("user2", "user2-123")))
+		leaseResp, err := user1c.Grant(ctx, 90)
+		testutil.AssertNil(t, err)
+		leaseID := leaseResp.ID
+		// permission of k2 is also granted to user2
+		err = user1c.Put(ctx, "k2", "val", config.PutOptions{LeaseID: leaseID})
+		testutil.AssertNil(t, err)
+		_, err = user2c.Revoke(ctx, leaseID)
+		testutil.AssertNil(t, err)
+
+		leaseResp, err = user1c.Grant(ctx, 90)
+		testutil.AssertNil(t, err)
+		leaseID = leaseResp.ID
+		// permission of k1 isn't granted to user2
+		err = user1c.Put(ctx, "k1", "val", config.PutOptions{LeaseID: leaseID})
+		testutil.AssertNil(t, err)
+		_, err = user2c.Revoke(ctx, leaseID)
+		if err == nil || !strings.Contains(err.Error(), rpctypes.ErrPermissionDenied.Error()) {
+			t.Fatalf("want %v error but got %v error", rpctypes.ErrPermissionDenied, err)
+		}
+	})
+}
+
+func TestAuthLeaseRevoke(t *testing.T) {
+	testRunner.BeforeTest(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1}))
+	defer clus.Close()
+	testutils.ExecuteUntil(ctx, t, func() {
+		rootAuthClient := testutils.MustClient(clus.Client(WithAuth("root", "root")))
+		authSetupDefaultTestUser(ctx, rootAuthClient, t)
+		// put with TTL 10 seconds and revoke
+		resp, err := rootAuthClient.Grant(ctx, 10)
+		if err != nil {
+			t.Fatal(err)
+		}
+		leaseID := resp.ID
+		if err = rootAuthClient.Put(ctx, "key", "val", config.PutOptions{LeaseID: leaseID}); err != nil {
+			t.Fatal(err)
+		}
+		if _, err = rootAuthClient.Revoke(ctx, leaseID); err != nil {
+			t.Fatal(err)
+		}
+		gresp, err := rootAuthClient.Get(ctx, "key", config.GetOptions{})
+		if err != nil || len(gresp.Kvs) != 0 {
+			t.Fatalf("want nil err and no kvs but got (%v) error and %d kvs", err, len(gresp.Kvs))
+		}
+	})
+}
+
+func TestAuthRoleGet(t *testing.T) {
+	testRunner.BeforeTest(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	clus := testRunner.NewCluster(ctx, t, config.WithClusterConfig(config.ClusterConfig{ClusterSize: 1}))
+	defer clus.Close()
+	cc := testutils.MustClient(clus.Client())
+	testutils.ExecuteUntil(ctx, t, func() {
+		authEnable(ctx, cc, t)
+		rootAuthClient := testutils.MustClient(clus.Client(WithAuth("root", "root")))
+		authSetupDefaultTestUser(ctx, rootAuthClient, t)
+		if _, err := rootAuthClient.RoleGet(ctx, "test-role"); err != nil {
```

Comment:
Original test validated here contents of the role, like what keys it can write and what keys it can read.

```
	expected := []string{
		"Role test-role",
		"KV Read:", "foo",
		"KV Write:", "foo",
	}
```